src/lib/builder.rs

```diff
@@ -0,0 +1,171 @@
+// Software as contributed to or provided by such licensor, or (ii) the Larger Works (as defined
+// below), to deal in both
+//
+// (a) the Software, and
+// (b) any piece of software and/or hardware listed in the lrgrwrks.txt file
+// if one is included with the Software (each a "Larger Work" to which the Software is contributed
+// by such licensors),
+//
+// without restriction, including without limitation the rights to copy, create derivative works
+// of, display, perform, and distribute the Software and make, use, sell, offer for sale, import,
+// export, have made, and have sold the Software and the Larger Work(s), and to sublicense the
+// foregoing rights on either these or other terms.
+//
+// This license is subject to the following condition: The above copyright notice and either this
+// complete permission notice or at a minimum a reference to the UPL must be included in all copies
+// or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING
+// BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+// NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+// DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+use std::collections::HashMap;
+use std::convert::{AsRef, TryFrom};
+use std::env::{current_dir, var};
+use std::error::Error;
+use std::fmt::Debug;
+use std::fs::{File, read_to_string};
+use std::io::Write;
+use std::path::{Path, PathBuf};
+
+use cfgrammar::yacc::{yacc_grm, YaccGrammar, YaccKind};
+use lrtable::{Minimiser, from_yacc, StateGraph, StateTable};
+use rmps::{Deserializer, Serializer};
+use serde::{Deserialize, Serialize};
+use typename::TypeName;
+
+const YACC_SUFFIX: &str = "_y";
+const YACC_FILE_EXT: &str = "y";
+const RUST_FILE_EXT: &str = "rs";
+
+/// Given the filename `x.y` as input, it will statically compile the file `src/x.y` into a Rust
+/// module which can then be imported using `lrpar_mod!(x_y)`. This is a convenience function
+/// around [`process_file`](fn.process_file.html) which makes it easier to compile `.y` files
+/// stored in a project's `src/` directory. Note that leaf names must be unique within a single
+/// project, even if they are in different directories: in other words, `a.y` and `x/a.y` will both
+/// be mapped to the same module `a_y` (and it is undefined what the resulting Rust module will
+/// contain).
+///
+/// # Panics
+///
+/// If the input filename does not end in `.y`.
+pub fn process_file_in_src<TokId>(srcp: &str)
+                               -> Result<(HashMap<String, TokId>), Box<Error>>
+                            where TokId: Copy + Debug + Eq + TryFrom<usize> + TypeName
+{
+    assert!(srcp.ends_with(YACC_FILE_EXT));
```

Comment:
Should this be a panic as in lrlex?

Reply:
Oops, it is below, but I forgot to remove the panic. Fixed.